File: leetcode/contains-duplicate.rs
/* -------------------------------------------------------------------------- */
/*              https://leetcode.com/problems/contains-duplicate              */
/* -------------------------------------------------------------------------- */

use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut hs = HashSet::new();
        for num in nums {
            if !hs.insert(num) {
                return true;
            }
        }
        return false;
    }
}

File: leetcode/cpp/contains-duplicate.cpp
/* -------------------------------------------------------------------------- */
/*              https://leetcode.com/problems/contains-duplicate              */
/* -------------------------------------------------------------------------- */

#include <iostream>
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {        
        // 1 liner
        return unordered_set<int>(nums.begin(), nums.end()).size() != nums.size();
    }
};